lib/item_repository.rb

```diff
@@ -16,9 +18,72 @@ def generate_items(filename)
     end
   end
 
-  def unit_price_to_dollars
+  def all
+    @items.length
+  end
+
+  def find_by_id(id)
+    @items.find do |item|
+      item.id == id.to_s
+    end
+  end
+
+  def find_by_name(name)
+    @items.find do |item|
+      item.name == name
+    end
+  end
+
+  def find_all_with_description(description)
+    @items.find_all do |item|
+      item.description.downcase == description.downcase
+    end
+  end
+
+  def find_all_by_price(price)
+    price_fix = "#{price}00"
```

Comment:
We might need to create a helper method for this one.

Reply:
Okay, we can take a look

Reply:
Excellent.